07-MemoryManagement/user_space/myprog.c

```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <time.h>
+#include <sys/time.h>
+#include <math.h>
+
+int i, *p;
+struct timespec ts;
+long before, relalloc, relfree;
+long allocsize[5];
+
+int main(void)
+{
+	allocsize[0] = (long)pow(2, 10);
+	allocsize[1] = (long)pow(2, 15);
+	allocsize[2] = (long)pow(2, 22);
+	allocsize[3] = (long)pow(2, 32);
+	allocsize[4] = (long)pow(2, 35);
+
+	printf("-----------------------------------------------------------\n");
+	printf("MALLOC\n");
+	printf("Buffer size(bytes)\tAllocation time(nS)\tFreeing time(nS)\n");
+	for (i = 0; i < 5; i++) {
+		clock_gettime(CLOCK_MONOTONIC, &ts);
+		before = ts.tv_nsec;
+		p = malloc(allocsize[i]);
+		if (p == NULL) {
+			printf("%20ld\t", (long)allocsize[i]);
+			printf("Malloc error\n");
+			continue;
+		}
+		clock_gettime(CLOCK_MONOTONIC, &ts);
+		relalloc = ts.tv_nsec - before;
+		before = ts.tv_nsec;
+		free(p);
+		clock_gettime(CLOCK_MONOTONIC, &ts);
+		relfree = ts.tv_nsec - before;
```

Comment:
Is this calculation of time duration correct?

Reply:
I think yes, because we use a high resolution timer.

Reply:
Is it possible to get negative values in relfree?

Reply:
Done. Now I take seconds.